lang/rust/rust.py

```diff
@@ -1,215 +1,522 @@
+        actions.edit.up()
 
-    def code_operator_bitwise_or():
-        actions.auto(' | ')
+    def code_comment_documentation_inner():
+        actions.insert('//! ')
 
-    def code_operator_bitwise_exclusive_or():
-        actions.auto(' ^ ')
+    def code_comment_documentation_block_inner():
+        actions.insert('/*!')
+        actions.key('enter')
+        actions.key('enter')
+        actions.insert('*/')
 
-    def code_operator_bitwise_left_shift():
-        actions.auto(' << ')
+    # tag: imperative
 
-    def code_operator_bitwise_left_shift_assignment():
-        actions.auto(' <<= ')
+    def code_block():
+        actions.insert('{}')
+        actions.key('left enter')
 
-    def code_operator_bitwise_right_shift():
-        actions.auto(' >> ')
+    def code_state_if():
+        actions.insert('if  {  }')
+        actions.key('left:5')
 
-    def code_operator_bitwise_right_shift_assignment():
-        actions.auto(' >>= ')
+    def code_state_else_if():
+        actions.insert(' else if  {  }')
+        actions.key('left:5')
 
-    def code_operator_object_accessor():
-        actions.auto('.')
+    def code_state_else():
+        actions.insert(' else {  }')
+        actions.key('left:2')
 
     def code_state_switch():
-        actions.insert('match ')
+        actions.insert('match  {  }')
+        actions.key('left:5')
 
-    def code_block():
-        actions.auto('{\n\n}')
-        actions.edit.left()
-        actions.edit.up()
-        actions.key('tab')
+    def code_state_for():
+        actions.insert('for  in  {\n}\n')
+        actions.key('up:2 end left:6')
 
-    def code_import():
-        actions.auto('use ')
+    def code_state_for_each():
+        actions.insert('for  in  {\n}\n')
+        actions.key('up:2 end left:6')
 
-    def code_comment_line_prefix():
-        actions.auto('//')
+    def code_state_while():
+        actions.insert('while  {\n}\n')
+        actions.key('up:2 end left:2')
 
-    def code_comment_documentation():
-        actions.auto('/// ')
+    def code_state_loop():
+        actions.insert('loop  {\n}\n')
+        actions.key('up:2 end')
+
+    def code_state_return():
+        actions.auto_insert('return ')
+
+    def code_break():
+        actions.auto_insert('break;')
+
+    def code_next():
+        actions.auto_insert('continue;')
+
+    # tag: object_oriented
+
+    def code_operator_object_accessor():
+        actions.auto_insert('.')
 
     def code_self():
-        actions.auto('self')
+        actions.auto_insert('self')
+
+    def code_define_class():
+        actions.auto_insert('struct ')
+
+    # tag: data_bool
 
     def code_insert_true():
-        actions.auto('true')
+        actions.auto_insert('true')
 
     def code_insert_false():
-        actions.auto('false')
+        actions.auto_insert('false')
 
-    def code_state_if():
-        actions.insert('if ')
+    # tag: data_null
+    # Convenience function, however, Option technically isn't null
```

Comment:
I had concerns about this in the original PR: https://github.com/knausj85/knausj_talon/pull/743#discussion_r790621760